documentation/src/main/asciidoc/topics/con_performance_data_consistency.adoc

```diff
@@ -0,0 +1,125 @@
+[id='performance-data-consistency_{context}']
+= Data consistency
+
+Data that resides on a distributed system is vulnerable to errors that can arise from temporary network outages, system failures, or just simple human error.
+These external factors are uncontrollable but can have serious consequences for quality of your data.
+The effects of data corruption range from lower customer satisfaction to costly system reconciliation that results in service unavailability.
+
+{brandname} can carry out ACID (atomicity, consistency, isolation, durability) transactions to ensure the cache state is consistent.
+
+Transactions are a sequence of atomic operations that {brandname} caries out as a single operation.
+Either all write operations in a transaction complete successfully or they all fail.
+In this way, the transaction either modifies the cache state in a consistent way, providing a history of reads and writes, or it does not modify cache state at all.
+
+The following is a simple transaction with one read and one write operation:
+
+[source,java,options="nowrap",subs=attributes+]
+----
+transaction.begin()
+cache.get(k1) // reads v1
+cache.put(k1, v2) // writes v2
+transaction.commit()
+----
+
+The main performance concern for enabling transactions is finding the balance between having a more consistent data set and increasing latency that degrades write throughput.
+
+[discrete]
```

Comment:
I still don't like the "locking mode" section... seems kind of out of context.
 Let me re-write it
```
## Locking mode

The incorrect locking mode may affect negatively the performance of your transactions. You should choose the locking mode which fits best with your workload.

For low contention workload (in other words, having 2 or more transactions write on the same key at the same time is rare), Optimistic Locking performs better. No penalty exists in the transaction execution but, at commit time, locks are acquired. If you don't have lock contention, the commit should be fast; but if you have contention, it may take time to acquire the lock and, unfortunately, all your work will be rolled back.

On other hand, for high contention workload, Pessimistic Locking will work better. Locks are acquired when the keys are accessed and it ensures no other transaction can modify them. The commit is done in a single phase since the locks are already acquired.

--- link to https://infinispan.org/docs/stable/titles/developing/developing.html#tx_locking
```